### ISSUE TITLE
setPreferredVideoMimeType/setPreferredVideoRoleFlags don't work for adaptive playbacks

### ISSUE THREAD

**Issue by ojw28** [google/ExoPlayer#9519]
As per title, the preferred video MIME type(s) don't seem to be used when making adaptive video track selections.

I found this when investigating https://github.com/google/ExoPlayer/issues/9463, where it would have been useful to provide the following as a viable workaround to prefer the H264 adaptation set:

```
trackSelector.setParameters(
    trackSelector
        .getParameters()
        .buildUpon()
        .setPreferredVideoMimeType(MimeTypes.VIDEO_H264));
```

**Comment by tonihei** [comment]
[Solution tracked internally by b/202380939]

**Comment by tonihei** [comment]
Note: The problem occurs for preferred video role flags, which are also only applied when making a fixed video track selection at the moment.

**Comment by rohitninawe** [comment]
is there any fix for [#9463 ] ?
